pytorch_lightning/trainer/progress.py

```diff
@@ -0,0 +1,73 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+
+
+@dataclass
+class LoopProgress:
+    """Basic dataclass to track loop progress during execution.
+
+    Args:
+        total_epochs_processed: Number of passes through the dataset as defined by the loop
+        total_batches_processed: Number of batches seen. This monotonically increases throughout execution
+        batches_processed_this_epoch: This is reset at the end of the epoch
+    """
+    total_epochs_processed: int = 0
+    total_batches_processed: int = 0
+    batches_processed_this_epoch: int = 0
```

Comment:
do we really need the _processed if it in all vars and whenever you use the attribute becomes very long and yield in not nice like breaking (due to formating)
```suggestion
    total_epochs: int = 0
    total_batches: int = 0
    batches_this_epoch: int = 0
```

Reply:
@Borda - sorry for the delay. I want to be as explicit about the naming as possible. for example, there are also fields like this: https://github.com/PyTorchLightning/pytorch-lightning/blob/29357ba94eea0af101a41553fe82d71630af3c78/pytorch_lightning/trainer/data_loading.py#L47

from the name alone, it's impossible to tell what `num_training_batches` refers to. is it the batches processed? this epoch? for the whole training? only from reading through the code can one say that its an estimation based on the dataloader length and limit train batches per epoch, but that it can also be infinite.

my hope for `processed` is that its clear that we've read the batch and run the step on it. this is also terminology i've seen used elsewhere to good effect